### PR TITLE
[SPARK-44320][SQL] Assign names to the error class _LEGACY_ERROR_TEMP_[1067,1150,1220,1265,1277]

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
The pr aims to assign names to the error class, include: 
- _LEGACY_ERROR_TEMP_1067 => UNSUPPORTED_FEATURE.DROP_DATABASE
- _LEGACY_ERROR_TEMP_1150 => UNSUPPORTED_DATA_TYPE_FOR_DATASOURCE
- _LEGACY_ERROR_TEMP_1220 => UNSUPPORTED_FEATURE.HIVE_TABLE_TYPE
- _LEGACY_ERROR_TEMP_1265 => LOAD_DATA_PATH_NOT_EXISTS
- _LEGACY_ERROR_TEMP_1277 => CREATE_VIEW_COLUMN_ARITY_MISMATCH.TOO_MANY_SOURCE_COLUMNS / CREATE_VIEW_COLUMN_ARITY_MISMATCH.NOT_ENOUGH_SOURCE_COLUMNS

### Why are the changes needed?
The changes improve the error framework.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
- Update & Add new UT.
- Manually test.
- Pass GA.

### CODE REVIEW

File: common/utils/src/main/resources/error/error-classes.json [hunk 1]
```diff
@@ -1204,7 +1225,7 @@
       },
       "MISMATCH_INPUT" : {
         "message" : [
-          "The input <inputType> '<input>' does not match the format."
+          "The input <inputType> <input> does not match the format."
```
Comment (panbingkun) [inline common/utils/src/main/resources/error/error-classes.json:1228]: This is the issue left by [SPARK-44299](https://github.com/apache/spark/pull/41858)
https://github.com/apache/spark/pull/41858#discussion_r1254022894

Comment (MaxGekk) [reply common/utils/src/main/resources/error/error-classes.json:1271]: Thank you.

File: common/utils/src/main/resources/error/error-classes.json [hunk 2]
```diff
@@ -2544,6 +2570,11 @@
       "The direct query on files does not support the data source type: <className>. Please try a different data source type or consider using a different query method."
     ]
   },
+  "UNSUPPORTED_DATA_TYPE_FOR_DATASOURCE" : {
+    "message" : [
+      "The <format> datasource don't support column <columnName> with the type of <columnType>."
```
Comment (MaxGekk) [inline common/utils/src/main/resources/error/error-classes.json:2575]:
```suggestion
      "The <format> datasource doesn't support the column <columnName> of the type <columnType>."
```

**Review by MaxGekk — CHANGES_REQUESTED** [review]

File: common/utils/src/main/resources/error/error-classes.json [hunk 3]
```diff
@@ -370,6 +370,27 @@
     ],
     "sqlState" : "42710"
   },
+  "CREATE_VIEW_COLUMN_ARITY_MISMATCH" : {
```
Comment (MaxGekk) [inline common/utils/src/main/resources/error/error-classes.json:373]: Could you align this new error class with the existing `INSERT_COLUMN_ARITY_MISMATCH` and output columns?

Comment (panbingkun) [reply common/utils/src/main/resources/error/error-classes.json:373]: Done.

**Review by MaxGekk — APPROVED** [review]

**Comment by MaxGekk** [comment]
I believe the failure is not related to PR's changes:
```python
RUN: /__w/spark/spark/dev/connect-gen-protos.sh /tmp/tmpo7t_pdiv
...
subprocess.CalledProcessError: Command '['/__w/spark/spark/dev/connect-gen-protos.sh', '/tmp/tmpo7t_pdiv']' returned non-zero exit status 1.
Error: Process completed with exit code 1.
```

+1, LGTM. Merging to master.
Thank you, @panbingkun.

**Comment by panbingkun** [comment]
> I believe the failure is not related to PR's changes:
> 
> ```python
> RUN: /__w/spark/spark/dev/connect-gen-protos.sh /tmp/tmpo7t_pdiv
> ...
> subprocess.CalledProcessError: Command '['/__w/spark/spark/dev/connect-gen-protos.sh', '/tmp/tmpo7t_pdiv']' returned non-zero exit status 1.
> Error: Process completed with exit code 1.
> ```
> 
> +1, LGTM. Merging to master. Thank you, @panbingkun.

Yeah, it is not related to this PR, it is the problem of using the 'deprecated' method for the remote plugin of buf.

**Pull request closed without merging**